architecture/2024-08-controller-component-threat-model.md

```diff
@@ -0,0 +1,259 @@
+- **Health Checks**: Health checks are implemented to monitor the status and performance of the Controller component as a whole, which includes both controllers and the validating webhook. They ensure that the controllers and the webhook are functioning correctly and can trigger corrective actions if any issues are detected.
+
+### Clients
+
+In this section, we will discuss the different clients of the Controller component. Clients are systems that interact with the Controller component to trigger actions. Here are the clients of the Controller component:
+
+1. **Kubernetes API Server**: The primary client that interacts with the controller. It communicates with the validating webhook and the controllers in case of resource changes (e.g., creation, update, or deletion of a Recipe or Deployment) requested by another interactor (for example; a human running `kubectl` commands). The controllers watch for these calls and reconcile the state of the resources accordingly.
+1. **Health Check Probes**: Kubernetes itself can act as a client by performing health and readiness checks on the controller manager.
+1. **Metrics Scrapers**: If metrics are enabled, Prometheus or other monitoring tools can scrape metrics from the controller manager.
+
+## Trust Boundaries
+
+We have a few different trust boundaries for the Controller component:
+
+- **Kubernetes Cluster**: The overall environment where the Controller component operates and receives requests from the clients.
+- **Namespaces within the Cluster**: Logical partitions within the cluster to separate and isolate resources and workloads.
+
+The Controller component lives inside the `radius-system` namespace in the Kubernetes cluster where it is installed. UCPD also resides within the same namespace.
+
+The Kubernetes API Server, which is the main interactor of the Controller component, runs in the `kube-system` namespace within the cluster.
+
+### Key Points of Namespaces
+
+1. **Isolation of Resources and Workloads**: Different namespaces separate and isolate resources and workloads within the Kubernetes cluster.
+2. **Access Controls and Permissions**: Access controls and other permissions are implemented to manage interactions between namespaces.
+3. **Separation of Concerns**: Namespaces support the separation of concerns by allowing different teams or applications to manage their resources independently, reducing the risk of configuration errors and unauthorized changes.
```

Comment:
This is much better, thank you. I think we could still be more specific about what "isolation" means. Isolation how?

I think this is lower urgency than the threats and mitigiations.